### PR TITLE
Port GPT Text Generation Guide to Keras Core

### PR DESCRIPTION
**Pull request by shivance**
Closes https://github.com/keras-team/keras-nlp/issues/1147

cc: @mattdangerw

### CODE REVIEW

**Comment by shivance** [comment]
This PR is WIP
1. Still testing for Torch and JAX backend
2. Need to update `ipynb` and `md` files.

**Comment by sachinprasadhs** [comment]
Could you please change it to draft till you make all the changes. Thanks

File: examples/generative/text_generation_gpt.py [hunk 1]
```diff
@@ -39,7 +40,9 @@
 import os
 import keras_nlp
 import tensorflow as tf
-from tensorflow import keras
+import keras_core as keras
```
Comment (mattdangerw) [inline examples/generative/text_generation_gpt.py:43]: Let's add some text to the introduction. Something like

```
This examples uses [Keras Core](https://keras.io/keras_core/) to work in any of
`"tensorflow"`, `"jax"` or `"torch"`. Support for Keras Core is baked into
KerasNLP, simply change the `"KERAS_BACKEND"` environment variable to select
the backend of your choice. We select the tensorflow backend below, but feel free
to mix it up!
```

**Comment by shivance** [comment]
This PR is ready for review, could you  @fchollet ?

File: examples/generative/text_generation_gpt.py [hunk 2]
```diff
@@ -193,19 +206,10 @@ def preprocess(inputs):
 # Output.
 outputs = keras.layers.Dense(VOCAB_SIZE)(x)
 model = keras.Model(inputs=inputs, outputs=outputs)
-loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+loss_fn = keras.losses.SparseCategoricalCrossentropy(from_logits=True)
 perplexity = keras_nlp.metrics.Perplexity(from_logits=True, mask_token_id=0)
 model.compile(optimizer="adam", loss=loss_fn, metrics=[perplexity])
 
-"""
-Let's take a look at our model summary - a large majority of the
```
Comment (mattdangerw) [inline examples/generative/text_generation_gpt.py:201]: Why did we remove this section?

File: examples/generative/md/text_generation_gpt.md [hunk 3]
```diff
@@ -264,19 +252,31 @@ model.fit(train_ds, validation_data=val_ds, verbose=2, epochs=EPOCHS)
 <div class="k-default-codeblock">
 ```
 Epoch 1/6
-3169/3169 - 132s - loss: 4.5592 - perplexity: 95.8829 - val_loss: 4.1382 - val_perplexity: 63.2792 - 132s/epoch - 42ms/step
+
+/usr/local/lib/python3.10/dist-packages/keras_core/src/layers/layer.py:759: UserWarning: Layer 'position_embedding' (of type PositionEmbedding) was passed an input with a mask attached to it. However, this layer does not support masking and will therefore destroy the mask information. Downstream layers will not see the mask.
```
Comment (mattdangerw) [inline examples/generative/md/text_generation_gpt.md:256]: we might want to take a look at these warning, are they valid?

File: examples/generative/text_generation_gpt.py (hunk 2 above)
Comment (shivance) [reply examples/generative/text_generation_gpt.py:201]: When I run a generation script, it simply fails saying that console width is not enough to print summary.

Comment (mattdangerw) [reply examples/generative/text_generation_gpt.py:201]: Hmm, probably not a reason to cut this section. Do you have the error handy?
